src/components/segmented-control/segmented-control.jsx

```diff
@@ -5,46 +5,107 @@ import SegmentedControlStyles from './segmented-control-styles';
 class SegmentedControl extends React.PureComponent {
   constructor(props, context) {
     super(props, context);
-    this.state = { selected: props.value };
+    if (this.props.type === 'radio') {
+      this.state = {
+        selected: props.value,
+      };
+    } else if (this.props.type === 'checkbox') {
+      // If multiple children
+      if (this.props.children.reduce) {
+        this.state = this.props.children.reduce((previous, child) => ({
+          ...previous,
+          [child.props.value]: !!child.props.checked,
+        }), {});
+      } else {
+        const childProps = this.props.children.props;
+        this.state = {
+          [childProps.value]: !!childProps.checked,
+        };
+      }
+    }
+    this.renderChild = this.renderChild.bind(this);
     this.handleChange = this.handleChange.bind(this);
+    this.childSelectedState = this.childSelectedState.bind(this);
     this.classes = this.context.styleManager.render(SegmentedControlStyles);
   }
 
+  childSelectedState() {
+    // Multiple children
+    if (this.props.children.reduce) {
+      if (this.props.type === 'radio') {
+        return this.props.children.reduce((previous, child) => ({
+          ...previous,
+          [child.props.value]: this.state.selected === child.props.value,
+        }), {});
+      } else if (this.props.type === 'checkbox') {
+        return this.props.children.reduce((previous, child) => ({
+          ...previous,
+          [child.props.value]: this.state[child.props.value],
+        }), {});
+      }
+    } else {
+      // Single child, only works with 'checkbox' behavior
+      const childValue = this.props.children.props.value;
+      return {
+        [childValue]: this.state[childValue],
+      };
+    }
+    return null;
+  }
+
   handleChange(e) {
     const index = e.currentTarget.value;
-    this.setState({
-      selected: index,
-    }, () => {
-      if (this.props.onChange) {
-        this.props.onChange(index);
-      }
+    if (this.props.type === 'radio') {
+      this.setState({
+        selected: index,
+      }, () => {
+        if (this.props.onChange) {
+          this.props.onChange(this.childSelectedState());
+        }
```

Comment:
```
if (this.props.onChange) {
    this.props.onChange(this.childSelectedState());
}
```
This can be moved down since it is the same for both cases